### PR TITLE
Preserving 64 bit floats for WebGL 2D plotting

### PR DESCRIPTION
**Pull request by monfera**
Solution for https://github.com/plotly/plotly.js/issues/995

Rebased on https://github.com/plotly/plotly.js/issues/413 to allow testing with a date axis.

For non-fancy `scattergl`, this is also needed: https://github.com/gl-vis/gl-scatter2d/pull/1

For fancy `scattergl`, this is also needed: 
- markers: https://github.com/gl-vis/gl-scatter2d-fancy/pull/2
- lines, line elbows (miters) and area shading: https://github.com/gl-vis/gl-line2d/pull/1
- error bars: https://github.com/gl-vis/gl-error2d/pull/1

### CODE REVIEW

File: src/lib/float32_truncate.js [hunk 1]
```diff
@@ -8,14 +8,25 @@
...
+ * For some reason, ES2015 Float32Array.prototype.slice takes
+ * 2x as long, therefore we aren't checking for its existence
  */
-module.exports = function truncate(float32ArrayIn, len) {
-    // for some reason, ES2015 Float32Array.prototype.slice takes 2x as long...
-    // therefore we aren't checking for its existence
-    var float32ArrayOut = new Float32Array(len);
-    for(var i = 0; i < len; i++) float32ArrayOut[i] = float32ArrayIn[i];
-    return float32ArrayOut;
+module.exports = function truncate(arrayIn, len) {
```
Comment (etpinard) [inline src/lib/float32_truncate.js]: Nicely done here.

Let's change the file name though to `typed_array_truncate.js` (or something better?))

**Comment by etpinard** [comment]
@monfera amazing work :tada: 

I've been playing with `gl2d` plots in the test dashboard for about 30 minutes now and I can't find any regressions. 

Go ahead and merge + publish all the gl-vis PRs. Once that's done, you'll need to rebase off `master` for the gl2d image test to pass on CircleCI (post https://github.com/plotly/plotly.js/pull/1067).

Then, we should modify the existing `gl2d_date_axes` mock and update its baseline to showcase the new high-precision logic.

Thanks again for your efforts :+1:

**Comment by monfera** [comment]
@etpinard all the above steps are now done except the update to `gl2d_date_axes.json`. Now it uses a daily raster. Should I just change it to a millisecond raster, and also add some points for around 1970 (for testing that it can do milliseconds even if the temporal domain of the data spans decades)?

**Comment by etpinard** [comment]
> Should I just change it to a millisecond raster, and also add some points for around 1970 (for testing that it can do milliseconds even if the temporal domain of the data spans decades)?

yes please!

**Comment by etpinard** [comment]
@monfera looks like the updated gl2d dates mock failed to generate on CircleCI - see [here](https://2847-45646037-gh.circle-artifacts.com/0/home/ubuntu/plotly.js/build/test_images/gl2d_date_axes.png).

I wonder if CircleCi doesn't like `Float64Array`s - which may be bad news for plot.ly's image server ...

**Pull request merged**